Expand settings

selfire1 opened the pull request:
New in this pull request:

Settings:
* Toggle folders in link scheme
* Set a Bible folder
* Set a custom scheme

Other:
* Lay the foundation for a custom scheme working

Errors:
* See [Define fixBibleReferences inside MyPlugin class · Issue #9](https://github.com/evantravers/obsidian-bible-references/issues/9)

This includes some significant changes, I hope it all makes sense. Let me know if there is anything I need to change!

selfire1 committed:
add multiple settings

selfire1 committed:
edit pathtofolder

selfire1 committed:
wikiBible conditional on settings (error)

selfire1 committed:
simplified if (thanks emile!)

selfire1 committed:
simplify if (thanks emile!)

selfire1 committed:
enable toggl, allow custom link pattern

selfire1 committed:
rename to customLinkScheme

selfire1 committed:
extra documentation

evantravers commented on main.ts:
```diff
@@ -104,7 +106,41 @@ function fixBibleReferences(app) {
+     
+    // There is surely a prettier way to do this – maybe objects?
+    let toReplace = ["{{book}}", "{{chapter}}", "{{verse}}", "{{input}}"]
+    let replaceWith = [`${b.book}`, `${b.chapter}`, `${b.verse}`, `${b.ref}`]
```
As I mentioned in the loom… you probably can just use

```typescript
let replaceWith = [b.book, b.chapter, b.verse, b.ref]
```

b.book is a string (like `"Genesis"`) so wrapping it in another template string where it echoes itself is unneeded.

Man this is all exciting! Great work!

evantravers commented on main.ts:
```diff
@@ -104,7 +106,41 @@ function fixBibleReferences(app) {
+      }
+      // Replace custom patterns with the variables
+      var linkPattern:any = linkPattern.replaceAll(toReplace[i], replaceWith[i])
+    }
```
Nice! There may be a cleaner way to do this, but as my professor used to say:

1. Make it work.
2. Make it nice.
3. Make it fast.

evantravers committed:
Ignore data.json

data.json will be particular to every install of the plugin... every
user will have different settings. By adding it to the .gitignore,
people can update the plugin and still have private settings.

evantravers committed:
MyPlugin -> BibleReferences

evantravers committed:
Clean trailing spaces

evantravers committed:
Change title

evantravers committed:
Change name of command

In the command window, the command is prefixed by the plugin name

evantravers committed:
Use mustache to render the link

This is still broken, and this whole branch is considered a WIP.

It does compile. It does render. It just needs some love.

evantravers commented:
So I've been tinkering around on a branch upstream of your branch: https://github.com/evantravers/obsidian-bible-references/commit/67d16dea47d5c229a7a36d1f4d8488c7921fa559

I'm using mustache to render the link. My hunch is that we don't have to have the folder settings, the user can just input it in the link format.

`[[ESV/{{book}}/{{book}}-{{chapter}}]]`

for instance.

It's a WIP, but you can check it out.

evantravers committed:
Automatically wrap the link

evantravers committed:
Make the text area bigger

Stealing this pattern from

https://github.com/hadynz/obsidian-kindle-plugin/blob/master/src/settingsTab/index.ts#L127-L129

This is the only example I had in my obsidian. :P

selfire1 committed:
add '}'

selfire1 committed:
add obsidian module

as per https://github.com/valentine195/obsidian-dice-roller/blob/115a505854636a19f901a8c4f39fe8bbb6a9b796/src/main.ts#L40

selfire1 committed:
add mustache

selfire1 committed:
call 'plugin' rather than globalThis
